### PR TITLE
Merge 'Development' branch into 'Main' branch - Datalink 2.0.0

### PR DESCRIPTION
**Pull request by 4x8Matrix**
# Context
Datalink has, and had needed a large update refactoring the technology being used; the source code behind Datalink being all too a bit too obscure and had no designated style-guide. 

The goal of this PR is to introduce a layer of understandability into the source, enabling those who want to contribute or add in features, to do so (or suggest them — We will listen!)

---

I'd go as far as saying this is the official 2.0.0 for Datalink; Datalink 1.0.0 is classed as the Prototype version, used for testing, regulating and essentially helping us developers find our feet.

# This PR
**Please refer to the 'Commit-History' feature as this PR is very large**

### CODE REVIEW

**Comment by CompeyDev** [comment]
Please follow [this](https://gist.github.com/5fc875a6e45844040ad9d9a3215f94e5) style guide, as the project will need to be bootstrapped with roblox-ts once again. The current style guide applies to lua only. 

Also, you might want to sign your commits next time onwards using GPG before pushing them to git.

**Comment by CompeyDev** [comment]
**Do not** merge this PR, as I will be setting my TypeScript compatibility soon.

File: .github/workflows/Selene.yml [hunk 1]
```diff
@@ -1,23 +1,27 @@
 name: Selene Linter
```
Comment (CompeyDev) [inline .github/workflows/Selene.yml:1]: Will be redoing the linter, because the End of Lines seem corrupted.

**Review by CompeyDev — APPROVED** [review]
Looks good, I'll make some changes and we'll be on our way to merge.

**Comment by 4x8Matrix** [comment]
> Please follow [this](https://gist.github.com/5fc875a6e45844040ad9d9a3215f94e5) style guide, as the project will need to be bootstrapped with roblox-ts once again. The current style guide applies to lua only.

I disagree, you should try to follow the [Roblox LuaU styleguide](https://roblox.github.io/lua-style-guide/) as that is the core languages of this SDK, me using typescript favors' the majority which shouldn't be the case. 

That said ~ the StyleGuide linked is also not documented well as it covers two subjects..? 

> Also, you might want to sign your commits next time onwards using GPG before pushing them to git.

My bad, moved to arch so my GPG keys got invalidated, will update them soon! 

> Do not merge this PR, as I will be setting my TypeScript compatibility soon.

Thats fine, please don't approve until you're happy the PR can be merged.

**Comment by howmanysmall** [comment]
🤓🤓🤓🤓🤓🤓🤓🤓🤓🤓🤓
NerdEmojiNerdGIF
🤓🤓🤓🤓🤓🤓🤓🤓🤓🤓

**Review by CompeyDev — APPROVED** [review]

**Review by CompeyDev — APPROVED** [review]

**Comment by CompeyDev** [comment]
Everything is okay, however we should really consider using the TypeScript Style Guide since this repo is linted and managed primarily by `roblox-ts`.

**Review by CompeyDev — APPROVED** [review]

**Review by CompeyDev — APPROVED** [review]
Ooh, VSCode tasks... That must be helpful for new contributors!

**Pull request closed without merging**